doc/website/getting-started/overview.md

```diff
@@ -0,0 +1,565 @@
+    if successful assign it a new value. Note that ``ptr`` points to an already default constructed sample, so we
+    cannot treat it as uninitialized memory and therefore must assign the data to send.
+    
+    If you are only using a simple data type which does not rely on RAII, you can also use the pointer to construct
+    the data via placement new instead.
+
+    ```cpp
+    new (ptr) CounterTopic(73);
+    ```
+
+3. Publish by copy
+
+    ```cpp
+    CounterTopic counter(73);
+    publisher.publishCopyOf(counter);
+    ```
+    This copies the constructed ``counter`` object and hence should mostly be used for small data.
+
+4. Publish the result of a computation
+
+    ```cpp
+    auto myComputation = [](CounterTopic* data) { *data = CounterTopic(73); };
+    auto result = publisher.publishResultOf(myComputation);
+    if (result.has_error())
+    {
+        // handle the error
+    }    
+    ```
+
+    This can be used if we want to set the data by some callable (i.e. lambda, function or [functor](https://en.wikipedia.org/wiki/Function_object#In_C_and_C++)). As with all the
+    other ways, it can fail when there is no memory for the sample availabe and this failure must be handled.
+
+#### Creating a subscriber
+
+We now create a corresponding subscriber, usually in another application. While it will work in the same application as
+well, there usually is no need sending it via the middleware in such a case.
+
+```cpp
+iox::popo::TypedSubscriber<CounterTopic> subscriber({"Group", "Instance", "CounterTopic"});
+subscriber.subscribe();
+```
+The template data type and the three string identifiers have to match those of the publisher, in other words the
+service descriptions have to be the same (otherwise we will not receive data from our publisher).
+
+We immediately subscribe here, but this can be postponed to the point where we actually want to receive data.
+
+#### Receiving data
+For simplicity we assume that we periodically check for new data. It is also possible to explicitly wait for data
+using the [Waitset](../../../iceoryx_examples/waitset/README.md). The code to receive the data is the same, the only
+difference is the way we wake up before checking for data.
+```cpp
+while (keepRunning)
+{
+    // wait for new data (either sleep and wake up periodically or by notification from the waitset)
+
+    subscriber->take()
+        .and_then([](iox::popo::Sample<const CounterTopic>& sample) {
+            CounterTopic* ptr = sample.get();
+            /* process the received data using the ptr */
+
+            /* alternatively use operator-> */
+            uint32_t counter = sample->counter;
+        })
+        .if_empty([] { /* no data received but also no error */ })
+        .or_else([](iox::popo::ChunkReceiveError) { /* handle the error */ });
+}
+
+```
+
+By calling ``take`` we get a ``iox::cxx::expected<iox::optional<iox::popo::Sample<const CounterTopic>>>``. Since this
```

Comment:
Needs to be discussed, we cannot distinguish between no data and other failures then. Might be worth it to have a dedicated return type here (a specialized expected, if you will). Cannot change anything here because it is just documenting the API as it is now.

Reply:
Added to https://github.com/eclipse/iceoryx/issues/408#issue-758449131

Reply:
1.0 API topic